R/mcmc.R

```diff
@@ -1,280 +1,151 @@
-                  rSeed <- abs(rSeed[-c(1:2)][rSeed[2]])
-
-                  ## specify the JAGS model
-                  jagsModel <-
-                      rjags::jags.model(file=modelFileName,
-                                        data=c(requiredData,
-                                               modelspecs),
-                                        inits=
-                                        ## add the RNG seed to the inits list:
-                                        ## (use Mersenne Twister as per R
-                                        ## default)
-                                        c(inits,
-                                          list(.RNG.name="base::Mersenne-Twister",
-                                               .RNG.seed=rSeed)),
-                                        quiet=!verbose,
-                                        ## important for
-                                        ## reproducibility:
-                                        n.adapt=0)
-
-                  ## run for the burnin time -> but don't show progress bar for
-                  ## this one in order not to confuse the user
-                  update(jagsModel,
-                         n.iter=options@burnin,
-                         progress.bar="none")
-
-                  ## afterwards generate more samples and save every step one
-                  ## code is:
-                  samplesCode <- "samples <-
-                      rjags::jags.samples(model=jagsModel,
-                                          variable.names=model@sample,
-                                          n.iter=
-                                              (options@iterations - options@burnin),
-                                          thin=options@step,
-                                          progress.bar=
-                                              ifelse(verbose,
-                                                     'text',
-                                                     'none'))"
-
-                  ## evaluate with or without outstream capturing
-                  if(verbose)
-                  {
-                      eval(parse(text=samplesCode))
-                  } else {
-
-                      ## this is necessary because some outputs
-                      ## are written directly from the JAGS compiled
-                      ## code to the outstream
-                      capture.output(eval(parse(text=samplesCode)))
-                  }
+setMethod(
+  f = "mcmc",
+  signature = signature(
+    data = "GeneralData",
+    model = "GeneralModel",
+    options = "McmcOptions"
+  ),
+  def = function(data,
+                 model,
+                 options,
+                 verbose = FALSE,
+                 from_prior = data@nObs == 0L,
+                 ...) {
+    assert_flag(verbose)
+    assert_flag(from_prior)
+
+    if (verbose) {
+      futile.logger::flog.threshold(futile.logger::INFO, name = "mcmc")
```

Comment:
I think we should get rid of the function-wise `verbose` arguments,because this is for me the main value of using logging package. instead we can provide a helper that sets the logging threshold globally (e.g. using `flog.threshold`).